### PR TITLE
Add package cpp-lazy

### PR DESCRIPTION
**Pull request by MarcDirven**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
https://github.com/xmake-io/xmake-repo/issues/1814

### CODE REVIEW

File: packages/c/cpp-lazy/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package("cpp-lazy")
+    set_homepage("https://github.com/MarcDirven/cpp-lazy")
+    set_description("A fast C++11/14/17/20 header only library for lazy evaluation and function tools")
+    
+    set_urls("https://github.com/MarcDirven/cpp-lazy/archive/refs/tags/v$(version).zip")
+
+    add_versions("7.0.2", "aced554a828ab8d2c426b9a9cc0a0d576752d83c5a8e77a28506575126c3e40e")
```
Comment (waruqi) [inline packages/c/cpp-lazy/xmake.lua:7]:
```lua
    set_urls("https://github.com/MarcDirven/cpp-lazy/archive/refs/tags/$(version).tar.gz",
             "https://github.com/MarcDirven/cpp-lazy.git")

    add_versions("v7.0.2", "7a5c2a42ce5c98343676d09761959b9821ec125dbff7b9f2028792c117de0b09")
```

Comment (waruqi) [reply packages/c/cpp-lazy/xmake.lua:7]: use tar.gz, it's smaller.

File: packages/c/cpp-lazy/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        import("package.tools.xmake").install(package)
+    end)
+
+    on_test(function(package)
+        assert(package:check_cxxsnippets({test = [[
+        void test() {
+            int arr[]{1, 2, 3, 4};
+            auto mapper{lz::map(arr, [](const int i) { return i + 1; })};
+        }
+    ]]}, {includes = "Lz/Map.hpp"}))
```
Comment (waruqi) [inline packages/c/cpp-lazy/xmake.lua:21]:
```
{includes = "Lz/Map.hpp", configs = {languages = "c++11"}}
```

Comment (MarcDirven) [reply packages/c/cpp-lazy/xmake.lua:21]: MSVC compiler doesn't support C++11, how can I check if the compiler supports C++11?

Comment (waruqi) [reply packages/c/cpp-lazy/xmake.lua:21]: msvc should support it, xmake will ignore it if this flag is not supported.

**Pull request merged**